plugin/types.go

```diff
@@ -193,3 +193,12 @@ type Readable interface {
 	Entry
 	Open(context.Context) (SizedReader, error)
 }
+
+// Deletable is an entry that can be deleted. Entries that implement Delete
+// should ensure that it and all its children are removed. If the entry has
+// any dependencies that need to be deleted, then Delete should return an
+// error.
+type Deletable interface {
+	Entry
+	Delete(context.Context) error
```

Comment:
Deleting an EC2 instance is equivalent to terminating it. According to https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/terminating-instances.html, termination is irreversible. However, terminated EC2 instances do show up in the AWS console for an hour.

Reason I bring this up is b/c if we delete an EC2 instance in Wash, that EC2 instance will disappear from the `instances` directory (breaking consistency with the AWS console). Is that OK? If so, we should probably filter out terminated EC2 instances when listing the `instances` directory.

Reply:
I lean towards not implementing `Delete` for EC2 instances and maybe outlining the reason in `docs <ec2_instance>`.

Reply:
I feel like this might be a common behavior. Deleting things may not be instantaneous, and we need to come up with some way to reflect that they're being removed.

Reply:
I think EC2 should respond to Delete, same as GCP. In each case, it looks like they may go through a TERMINATED state before being removed. I think for now, we should consider TERMINATED as absent and not list those instances. The Delete implementation for them should only return once they've arrived at the TERMINATED state.

The API should trigger Delete on batched instances in parallel so that waiting on multiple to go through that lifecycle doesn't take too long.